### PR TITLE
show rustup version in settings panel

### PR DESCRIPTION
**Pull request by zjhmale**
First step for #308 

image

seems to achieve the whole supporting is a long journey, so maybe tackle it with seperated PRs is better 😃

### CODE REVIEW

File: src/main/kotlin/org/rust/cargo/toolchain/RustToolchain.kt [hunk 1]
```diff
@@ -176,6 +188,23 @@ private fun parseRustcVersion(lines: List<String>): Version? {
...
+    //  ```
+    //  rustup 0.5.0 (4be1012 2016-07-30)
+    //  ```
+    val releaseRe = """rustup (\d+)\.(\d+)\.(\d+) \(.*\)""".toRegex()
+
+    val match = lines.mapNotNull { releaseRe.matchEntire(it) }.firstOrNull() ?: return null
+
+    val major = match.groups[1]!!.value.toInt()
+    val minor = match.groups[2]!!.value.toInt()
+    val build = match.groups[3]!!.value.toInt()
```
Comment (mkaput) [inline src/main/kotlin/org/rust/cargo/toolchain/RustToolchain.kt]: What if user has `rustup` binary which isn't rustup at all? I mean this will fail when output doesn't match regexp.

Comment (zjhmale) [inline src/main/kotlin/org/rust/cargo/toolchain/RustToolchain.kt]: if the output doesn't match regexp, it will return null for `Version?` and show `N/A` on the panel just like `rustc` and `cargo`.

Comment (mkaput) [inline src/main/kotlin/org/rust/cargo/toolchain/RustToolchain.kt]: Uh, must have to better get into regexp api then 😅

File: src/main/kotlin/org/rust/cargo/toolchain/RustToolchain.kt [hunk 2]
```diff
@@ -14,8 +14,8 @@ import java.io.File
 
 data class RustToolchain(val location: String) {
 
-    fun looksLikeValidToolchain(): Boolean =
-        File(pathToExecutable(CARGO)).canExecute()
+    fun looksLikeValidToolchain(exec: String = CARGO): Boolean =
```
Comment (matklad) [inline src/main/kotlin/org/rust/cargo/toolchain/RustToolchain.kt]: Let's not add this `exec` parameter and instead create `hasExecuteable(exec: String)` *private* helper which can be used by both `looksLikeValidToolchain` and `queryRustupVersion`

File: src/main/kotlin/org/rust/cargo/toolchain/RustToolchain.kt [hunk 3]
```diff
@@ -14,8 +14,11 @@ import java.io.File
 
 data class RustToolchain(val location: String) {
 
+    private fun hasExecuteable(exec: String): Boolean =
```
Comment (zjhmale) [inline src/main/kotlin/org/rust/cargo/toolchain/RustToolchain.kt]: hope not misunderstand your purpose here :sweat_smile:

Comment (matklad) [inline src/main/kotlin/org/rust/cargo/toolchain/RustToolchain.kt]: Yes, this is exactly what I had in mind! 

And now that this is written this way, I see that the old code has probably had a bug, because it did check for the presence of `cargo` and not `rustc` in `queryRustcVersion`!

Comment (zjhmale) [inline src/main/kotlin/org/rust/cargo/toolchain/RustToolchain.kt]: @matklad should i make another PR for this?

Comment (matklad) [inline src/main/kotlin/org/rust/cargo/toolchain/RustToolchain.kt]: No, I am already auditing the usages of `looksLikeValidToolchains` myself :)

Comment (zjhmale) [inline src/main/kotlin/org/rust/cargo/toolchain/RustToolchain.kt]: cool 😄
